tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -52,59 +52,69 @@ object CounterexampleWriter {
 }
 
 class TlaCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter {
-
-  def printStateFormula(pretty: PrettyWriter, state: State) = {
+  def printStateFormula(pretty: PrettyWriter, state: State): Unit = {
     if (state.isEmpty) {
       pretty.write(ValEx(TlaBool(true)))
+      pretty.writeln()
     } else {
       state.toList.sortBy(_._1).foreach { case (name, value) =>
-        writer.print(s"/\\ $name = ")
+        pretty.write(s"/\\ $name = ")
         pretty.write(value)
-        writer.println
+        pretty.writeln()
       }
     }
   }
 
   override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState]): Unit = {
     val pretty = new PrettyWriter(writer)
 
-    writer.println("%s MODULE counterexample %s\n".format("-" * 25, "-" * 25))
-    writer.println("EXTENDS %s\n".format(rootModule.name))
+    pretty.prettyWriteDoc(pretty.moduleNameDoc("counterexample", 25))
+    pretty.writeln()
+    pretty.prettyWriteDoc(pretty.moduleExtendsDoc(rootModule.name))
+    pretty.writeln()
 
     states.zipWithIndex.foreach {
       case (state, 0) =>
-        writer.println("(* Constant initialization state *)")
-        writer.println(s"\nStateConst ==")
+        pretty.writeln("(* Constant initialization state *)")
```

Comment:
I would object against introducing `writeln` in the pretty printer. Can you introduce a specialized method for writing comments instead? It could do some special processing of comments, like wrapping and indentation.

Reply:
Can you be more specific with exactly what kind of "special processing" you want?

Reply:
I just want to have a method `writeComment(text: String)` that wraps `text` into `(*` and `*)`. In the future, we may add smarter code. My main objection is that if you add `writeln`, it is a recepy for pushing unformatted text in the middle of the pretty printed text.